include/ofi_hmem.h

```diff
@@ -80,18 +80,19 @@ hsa_status_t ofi_hsa_amd_reg_dealloc_cb(void *ptr,
 
 #endif /* HAVE_ROCR */
 
-int rocr_memcpy(void *dest, const void *src, size_t size);
+int rocr_memcpy(void *dest, const void *src, uint64_t device, size_t size);
 int rocr_hmem_init(void);
 int rocr_hmem_cleanup(void);
 bool rocr_is_addr_valid(const void *addr);
 
-int cuda_copy_to_dev(void *dev, const void *host, size_t size);
-int cuda_copy_from_dev(void *host, const void *dev, size_t size);
+int cuda_copy_to_dev(void *dev, const void *host, uint64_t device, size_t size);
+int cuda_copy_from_dev(void *host, const void *dev, uint64_t device, size_t size);
 int cuda_hmem_init(void);
 int cuda_hmem_cleanup(void);
 bool cuda_is_addr_valid(const void *addr);
 
-static inline int ofi_memcpy(void *dest, const void *src, size_t size)
+static inline int ofi_memcpy(void *dest, const void *src, uint64_t device,
```

Comment:
Do the GPU APIs put the device identifier as the 3rd parameter?  This looks like an unnatural place to my tastes.

Reply:
The device identifier is just an index but is used to access which command queue to use which would be first in the ZE calls. I can move the the device before the pointers and be consistent.

Reply:
I would move it first or last or immediately before the buffer that it represents or somewhere that it doesn't look like an odd growth coming out of the middle of the standard memcpy API...